src/stripeJavaLanguageClient/utils.ts

```diff
@@ -0,0 +1,665 @@
+import * as fse from 'fs-extra';
+import * as glob from 'glob';
+import * as path from 'path';
+import {
+  ConfigurationTarget,
+  ExtensionContext,
+  OutputChannel,
+  RelativePattern,
+  TextDocument,
+  Uri,
+  WorkspaceConfiguration,
+  env,
+  window,
+  workspace,
+} from 'vscode';
+import {
+  Executable,
+  ExecutableOptions,
+  Location,
+  NotificationType,
+  RequestType,
+  TextDocumentPositionParams,
+} from 'vscode-languageclient';
+import javaPatterns from '../../config/api_ref/patterns_java.json';
+
+const expandHomeDir = require('expand-home-dir');
+
+const isWindows: boolean = process.platform.indexOf('win') === 0;
+const JAVAC_FILENAME = 'javac' + (isWindows ? '.exe' : '');
+const JAVA_FILENAME = 'java' + (isWindows ? '.exe' : '');
+
+export const ACTIVE_BUILD_TOOL_STATE = 'java.activeBuildTool';
+export const DEBUG_VSCODE_JAVA = 'DEBUG_VSCODE_JAVA';
+export const EXTENSION_NAME_STANDARD = 'stripeJavaLanguageServer (Standard)';
+export const EXTENSION_NAME_SYNTAX = 'stripeJavaLanguageServer (Syntax)';
+export const IS_WORKSPACE_JDK_ALLOWED = 'java.ls.isJdkAllowed';
+export const JDTLS_CLIENT_PORT = 'JDTLS_CLIENT_PORT';
+export const SYNTAXLS_CLIENT_PORT = 'SYNTAXLS_CLIENT_PORT';
+export const SERVER_PORT = 'SERVER_PORT';
+
+export interface JDKInfo {
+  javaHome: string;
+  javaVersion: number;
+}
+
+export interface StatusReport {
+  message: string;
+  type: string;
+}
+
+export enum ClientStatus {
+  Uninitialized = 'Uninitialized',
+  Initialized = 'Initialized',
+  Starting = 'Starting',
+  Started = 'Started',
+  Error = 'Error',
+  Stopping = 'Stopping',
+}
+
+export enum ServerMode {
+  STANDARD = 'Standard',
+  LIGHTWEIGHT = 'LightWeight',
+  HYBRID = 'Hybrid',
+}
+
+export namespace StatusNotification {
+  export const type = new NotificationType<StatusReport>('language/status');
+}
+
+export interface FindLinksParams {
+  type: string;
+  position: TextDocumentPositionParams;
+}
+
+export interface LinkLocation extends Location {
+  displayName: string;
+  kind: string;
+}
+
+export namespace FindLinks {
+  export const type = new RequestType<FindLinksParams, LinkLocation[], void>('java/findLinks');
+}
+
+export function getJavaConfiguration(): WorkspaceConfiguration {
+  return workspace.getConfiguration('java');
+}
+
+export async function ensureNoBuildToolConflicts(
+  context: ExtensionContext,
+  outputChannel: OutputChannel,
+): Promise<boolean> {
+  const isMavenEnabled: boolean =
+    getJavaConfiguration().get<boolean>('import.maven.enabled') || false;
+  const isGradleEnabled: boolean =
+    getJavaConfiguration().get<boolean>('import.gradle.enabled') || false;
+  if (isMavenEnabled && isGradleEnabled) {
+    const activeBuildTool: string | undefined = context.workspaceState.get(ACTIVE_BUILD_TOOL_STATE);
+    if (!activeBuildTool) {
+      if (!(await hasBuildToolConflicts())) {
+        return true;
+      }
+      outputChannel.appendLine('Build tool conflicts are detected in workspace.');
```

Comment:
What should the users do in this case?

Reply:
i will update the output to include user next step